### PR TITLE
ForgottenPassword: Move view to login screen

### LINKED ISSUE #24606: Update the look and feel of Forgot password view to match new design of the login page

### PR DESCRIPTION
**Pull request by tskarhed**
Peek 2020-06-08 14-29

**What this PR does / why we need it**:
- Creates LoginLayout so it can be used on multiple pages
- Updates the "Forgotten password" flow to new form styles and LoginLayout
- Misc. small UX updates for login
  - Full width Sign Up button
  - Smoother appear animation for the login forms
  - Forgotten password same color as all links

**Questions:**
- Does it make sense to not have the new view animated in after sending the email? I think the animation is there for context switch.
- Running the mail server remotely causes the app url in the sent email to generate the wrong port in the link. Is this something that anybody else notices?

Closes https://github.com/grafana/grafana/issues/24606

### CODE REVIEW

File: public/app/core/components/Login/LoginPage.tsx [hunk 1]
```diff
@@ -37,28 +39,41 @@ export const LoginPage: FC = () => {
             changePassword,
             skipPasswordChange,
             isChangingPassword,
+            resetCode,
           }) => (
             <div className={loginStyles.loginOuterBox}>
-              {!isChangingPassword && (
+              {!(isChangingPassword || forgotPassword || resetCode) && (
```
Comment (tskarhed) [inline public/app/core/components/Login/LoginPage.tsx:45]: If there is something more added here, it definitely has to be rewritten to some kind of routing solution.

File: public/app/core/components/Login/ForgottenPassword.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      setEmailSent(true);
+    }
+  };
+
+  if (emailSent) {
+    return (
+      <div>
+        <p>
+          An email with a reset link has been sent to the email address. <br />
+          You should receive it shortly.
```
Comment (dprokop) [inline public/app/core/components/Login/ForgottenPassword.tsx:24]: Think the `<br />` ere is not necessary. It looks weird in the UI

**Comment by dprokop** [comment]
@jessover9000 can you please take a look and test this? :)

**Review by dprokop — CHANGES_REQUESTED** [review]
One major thing that I find confusing with this proposal is the fact that password reset is handled on the same route as login. I go to login -> reset the password, go back, and... I'm not on a login page anymore, because it's the same route. I don't like it, but it might be my personal bias. 

I'm not convinced with this change, think we need to figure out a way to handle the route change to enable the back button. It may be tricky, and I would probably suggest keeping the "old" approach with a separate route (we can change it so something different than `/user/password/send-reset-email`). With react-router handling back will be easy-peasy.

Also, I'm not sure how long is the reset password link valid, but we need to add a redirect for users coming from old links, otherwise, they will encounter 404, don't they? 

On this branch going to http://localhost:3000/user/password/send-reset-email ends up in the old form.

Think the UI needs some more love too, like centering the `Back to login` button. I think it's worth having a quick pair programming with @jessover9000 to tweak the UI

**Review by jessover9000 — APPROVED** [review]
Did some quick-fix tweaks in a collab session with @tskarhed and for now I approve! We should probably rework the entire login form at some point though, the full-width buttons are not ideal from a UX standpoint.

**Review by mckn — APPROVED** [review]
Works as expected, great work!

approved

**Review by torkelo — APPROVED** [review]
Looks good! merge it!

**Comment by dprokop** [comment]
@tskarhed i see this:

image


macOS, Chrome 83.0.4103.106, 27'' screen

File: public/app/core/components/ForgottenPassword/ChangePassword.tsx [hunk 3]
```diff
@@ -1,10 +1,10 @@
 import React, { FC, SyntheticEvent } from 'react';
 import { Tooltip, Form, Field, Input, VerticalGroup, Button, LinkButton } from '@grafana/ui';
 import { selectors } from '@grafana/e2e-selectors';
-import { submitButton } from './LoginForm';
+import { submitButton } from '../Login/LoginForm';
 interface Props {
   onSubmit: (pw: string) => void;
-  onSkip: (event?: SyntheticEvent) => void;
+  onSkip?: (event?: SyntheticEvent) => void;
```
Comment (dprokop) [inline public/app/core/components/ForgottenPassword/ChangePassword.tsx:7]: Why is this optional?

File: public/app/core/components/ForgottenPassword/ForgottenPassword.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  userOrEmail: string;
+}
+
+const paragraphStyles = (theme: GrafanaTheme) => css`
+  color: ${theme.colors.formDescription};
+  font-size: ${theme.typography.size.sm};
+  font-weight: ${theme.typography.weight.regular};
+  margin-top: ${theme.spacing.sm};
+  display: block;
+`;
```
Comment (dprokop) [inline public/app/core/components/ForgottenPassword/ForgottenPassword.tsx:17]: `stylesFactory` maybe?

Comment (dprokop) [reply public/app/core/components/ForgottenPassword/ForgottenPassword.tsx:17]: Ah, I see `useStyles` now. I like it!

File: public/app/core/components/ForgottenPassword/ForgottenPassword.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+export const ForgottenPassword: FC = () => {
+  const [emailSent, setEmailSent] = useState(false);
+  const styles = useStyles(paragraphStyles);
+
+  const sendEmail = async (formModel: EmailDTO) => {
+    const res = await getBackendSrv().post('/api/user/password/send-reset-email', formModel);
+    if (res) {
+      setEmailSent(true);
+    }
+  };
```
Comment (dprokop) [inline public/app/core/components/ForgottenPassword/ForgottenPassword.tsx:28]: Think this can be moved outside of the component

**Review by dprokop — APPROVED** [review]
Looking good, found some nits :)

File: public/app/core/components/ForgottenPassword/ChangePassword.tsx (hunk 3 above)
Comment (tskarhed) [reply public/app/core/components/ForgottenPassword/ChangePassword.tsx:7]: You have the regular change password and the "admin-skip" change password. Could use children there instead and pass in a Skip button too. What do you think?

File: public/app/core/components/ForgottenPassword/ForgottenPassword.tsx (hunk 5 above)
Comment (tskarhed) [reply public/app/core/components/ForgottenPassword/ForgottenPassword.tsx:28]: Sure, but setEmailSent still needs to be in the component

File: public/app/core/components/Login/LoginLayout.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+export const getLoginStyles = (theme: GrafanaTheme) => {
+  return {
+    container: css`
+      min-height: 100vh;
+      background-position: center;
+      background-repeat: no-repeat;
+      min-width: 100%;
+      margin-left: 0;
+      background-color: $black;
```
Comment (Clarity-89) [inline public/app/core/components/Login/LoginLayout.tsx:54]: @tskarhed I think you've missed a var here.

File: public/app/core/components/Login/LoginLayout.tsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+      align-items: center;
+      justify-content: center;
+      flex-direction: column;
+      padding: ${theme.spacing.lg};
+    `,
+    titleWrapper: css`
+      text-align: center;
+    `,
+    mainTitle: css`
+      font-size: '32px';
```
Comment (Clarity-89) [inline public/app/core/components/Login/LoginLayout.tsx:79]: This shouldn't have quotes.

**Pull request merged**